src/helpers/genScreenshot.ts

```diff
@@ -0,0 +1,62 @@
+import chromium from 'chrome-aws-lambda'
+import puppeteerLib from 'puppeteer'
+
+export type DeviceModel = 'mobile' | 'desktop'
+
+export type IScreenShotOptions = {
+    mode?: DeviceModel,
+    width?: number,
+    height?: number,
+}
+
+async function getPuppeteerExcutable() {
+    const cal = await chromium.executablePath
+    if (!cal) return puppeteerLib
+    return chromium.puppeteer
+}
+
+async function getDevice(brand = 'iPhone X') {
+    const puppeteer = await getPuppeteerExcutable()
+    return puppeteer.devices[brand]
+}
+
+async function getBrowserInstance(width?: number, height?: number) {
+    // 1280, 720
+    const puppeteer = await getPuppeteerExcutable()
+    const browser = puppeteer.launch({
+        args: chromium.args,
+        headless: true,
+        defaultViewport: {
+            width: width || 1280,
+            height: height || 720
+        },
+        executablePath: await chromium.executablePath || undefined,
+        ignoreHTTPSErrors: true
+    })
+    return browser
+}
+
+export default async function genScreenshot(url: string, options: IScreenShotOptions): Promise<Buffer | undefined> {
+    console.time("screenshot")
+    const browser = await getBrowserInstance(options.width, options.height)
+
+    try {
+        const device = await getDevice()
+        device.viewport.width = options.width || device.viewport.width
+        device.viewport.height = options.height || device.viewport.height
+        let page = await browser.newPage()
+        if (options.mode == 'mobile') {
+            await page.emulate(device)
+        }
+        await page.goto(url)
```

Comment:
I believe it's not adequate here. It has to wait some content loading, or even more wait for some basic entry transition. Try theme [a robot named jimmy] with url `https://czg.vercel.app/theme/5486806e35e68e5e001940ff35b11b37`.

[https://stackoverflow.com/questions/46160929/puppeteer-wait-for-all-images-to-load-then-take-screenshot](https://stackoverflow.com/questions/46160929/puppeteer-wait-for-all-images-to-load-then-take-screenshot) might be helpful.

Reply:
@csszen maybe we can merge it first as it works for most circumstances.